logprep/connector/file/input.py

```diff
@@ -0,0 +1,346 @@
+                raise ValueError(f"Config attribute {attribute} needs to point to a existing file")
+
+        start: str = field(validator=validators.instance_of(str), default="begin")
+        """Defines the behaviour of the file monitor with the following options:
+        ``begin``: starts to read from the beginning of a file
+        ``end``: goes initially to the end of the file and waits for new content"""
+
+        @start.validator
+        def validate_start(self, attribute, value):
+            """Helper Function to validate the input values in config file"""
+            possible_values = ["begin", "end"]
+            if value not in possible_values:
+                raise ValueError(
+                    f"Config attribute {attribute} needs to be one of the \
+                            values in {possible_values}"
+                )
+
+        watch_file: str = field(validator=validators.instance_of(bool), default=True)
+        """Defines the behaviour of the file monitor with the following options:
+        ``True``: Read the file like defined in `start` param and monitor continuously
+        for newly appended log lines or file changes
+        ``False``: Read the file like defined in `start` param only once and exit afterwards"""
+
+        interval: int = field(default=1)
+        """Defines the refresh interval, how often the file is checked for changes"""
+
+        @interval.validator
+        def validate_interval(self, attribute, value):
+            """Helper Function to validate the input values in the config file"""
+            possible_types = [int, float]
+            if not type(value) in possible_types:
+                raise ValueError(
+                    f"Config attribute {attribute} needs to be onf of types in {possible_types}"
+                )
+            possible_range = [0.05, 100]
+            if not possible_range[0] <= value <= possible_range[1]:
+                raise ValueError(
+                    f"Config attribute {attribute} needs to be in range of the \
+                            values in {possible_range}"
+                )
+
+    def _calc_file_fingerprint(self, file_pointer, exist_fingerprint_size="") -> tuple:
+        """This function creates a crc32 fingerprint of the first 256 bytes of a given file
+        If the existing log file is less than 256 bytes, it will take what is there
+        and return also the size"""
+        if exist_fingerprint_size:
+            fingerprint_length = exist_fingerprint_size
+        else:
+            file_size = self._get_file_size(self._config.documents_path)
+            if 1 < file_size < 256:
+                fingerprint_length = file_size
+            else:
+                fingerprint_length = 256
+        fingerprint_bytes = str.encode(file_pointer.read(fingerprint_length))
+        crc32 = zlib.crc32(fingerprint_bytes) % 2**32
+        file_pointer.seek(0)
+        return crc32, fingerprint_length
```

Comment:
```suggestion
    def _calc_file_fingerprint(self, file_pointer, fingerprint_length="") -> tuple:
        """This function creates a crc32 fingerprint of the first 256 bytes of a given file
        If the existing log file is less than 256 bytes, it will take what is there
        and return also the size"""
        if not fingerprint_length:
            file_size = self._get_file_size(self._config.documents_path)
            if file_size > 1 and file_size < 256:
                fingerprint_length = file_size
            else:
                fingerprint_length = 256
        fingerprint_bytes = str.encode(file_pointer.read(fingerprint_length))
        crc32 = zlib.crc32(fingerprint_bytes) % 2**32
        file_pointer.seek(0)
        return crc32, fingerprint_length
```